### PR TITLE
Cleanup some tests

### PR DESCRIPTION
**Pull request by regiskuckaertz**
Not sure this is going to work, but I've done two things:
- replace d`fastdom.defer` calls with promise callback chains, I'm surprised the former ever worked
- disabled `Template` tests: firstly, the `done` callback was called _before_ the promise resolved, leaving dangling promises in the air. I moved it into the promise chain, but then it never got called, apparently because a call to `require` never returns. I will investigate that one later.

cc @TBonnin @jranks123

### CODE REVIEW

File: static/test/javascripts/spec/commercial/creatives/template.spec.js [hunk 1]
```diff
@@ -22,8 +22,8 @@ define([
 
     });
 
-    describe('Manual inline', function () {
-        beforeEach(function (done) {
+    xdescribe('Manual inline', function () {
```
Comment (TBonnin) [inline static/test/javascripts/spec/commercial/creatives/template.spec.js]: interesting way to disable tests. TIL 😁

**Comment by TBonnin** [comment]
I am not an expert but it lgtm 👍

File: static/src/javascripts/projects/commercial/modules/creatives/template.js [hunk 2]
```diff
@@ -97,10 +97,12 @@ define([
...
 
-                resolve(fastdom.write(function () {
+                fastdom.write(function () {
                     this.adSlot.insertAdjacentHTML('beforeend', creativeHtml);
-                    return true;
-                }, this));
+                }, this)
+                .then(function () {
+                    resolve(true);
+                });
```
Comment (jfsoul) [inline static/src/javascripts/projects/commercial/modules/creatives/template.js]: Does the `true` passed to resolve matter?  Would `.then(resolve)` work in this case?

Also: @gtrufitt this looks familiar!

Comment (gtrufitt) [inline static/src/javascripts/projects/commercial/modules/creatives/template.js]: 😆 Indeed. Hopefully it was this + the other bits!

File: static/test/javascripts/spec/commercial/creatives/template.spec.js [hunk 3]
```diff
@@ -46,12 +45,15 @@ define([
                 offer_button: 'yes',
                 clickMacro: '%%CLICK_URL_ESC%%'
             };
-            new Template(slot, params).create().then(function () {
-                expect(document.querySelector('.commercial', slot)).not.toBe(null);
-            });
+            new Template(slot, params).create()
+            .then(function () {
+                expect(document.querySelector('.adverts', slot)).not.toBe(null);
+            })
```
Comment (jfsoul) [inline static/test/javascripts/spec/commercial/creatives/template.spec.js]: how come the change of selector in this one?

**Comment by jfsoul** [comment]
looks good aside from minor comments 👍 

It won't fail for me locally any more (but I should hope not with the template tests disabled!)

File: static/test/javascripts/spec/commercial/creatives/template.spec.js (hunk 3 above)
Comment (regiskuckaertz) [inline static/test/javascripts/spec/commercial/creatives/template.spec.js]: The `.commercial` class does not exist anymore. I will update these tests later.

File: static/src/javascripts/projects/commercial/modules/creatives/template.js (hunk 2 above)
Comment (regiskuckaertz) [inline static/src/javascripts/projects/commercial/modules/creatives/template.js]: I could do that, yes. Nice.

**Comment by jfsoul** [comment]
And thanks for looking at this for us! 👏

**Comment by gtrufitt** [comment]
Indeed, cheers @regiskuckaertz 💯

**Pull request merged**